pkg/cmd/template/schema_author_test.go

```diff
@@ -756,6 +756,196 @@ schema.yml:
+`
+
+			filesToProcess := files.NewSortedFiles([]*files.File{
+				files.MustNewFileFromSource(files.NewBytesSource("schema.yml", []byte(schemaYAML))),
+			})
+
+			assertFails(t, filesToProcess, expectedErr, opts)
+		})
+		t.Run("has more than two args in an example", func(t *testing.T) {
+			schemaYAML := `#@data/values-schema
+---
+#@schema/examples ("key example", "val", "value")
+key: val
+`
+			expectedErr := `
+Invalid schema
+==============
+
+syntax error in @schema/examples annotation
+schema.yml:
+    |
+  3 | #@schema/examples ("key example", "val", "value")
+  4 | key: val
+    |
+
+    = found: 3-tuple argument in @schema/examples (by schema.yml:3)
+    = expected: 2-tuple containing description (string) and example value (of expected type)
+`
+
+			filesToProcess := files.NewSortedFiles([]*files.File{
+				files.MustNewFileFromSource(files.NewBytesSource("schema.yml", []byte(schemaYAML))),
+			})
+
+			assertFails(t, filesToProcess, expectedErr, opts)
+		})
+		t.Run("example description is not a string", func(t *testing.T) {
+			schemaYAML := `#@data/values-schema
+---
+#@schema/examples (3.14, 5)
+key: 7.3
+`
+			expectedErr := `
+Invalid schema
+==============
+
+syntax error in @schema/examples annotation
+schema.yml:
+    |
+  3 | #@schema/examples (3.14, 5)
+  4 | key: 7.3
+    |
+
+    = found: float value for @schema/examples (at schema.yml:3)
+    = expected: 2-tuple containing description (string) and example value (of expected type)
+`
+
+			filesToProcess := files.NewSortedFiles([]*files.File{
+				files.MustNewFileFromSource(files.NewBytesSource("schema.yml", []byte(schemaYAML))),
+			})
+
+			assertFails(t, filesToProcess, expectedErr, opts)
+		})
+		t.Run("is key=value pair", func(t *testing.T) {
+			schemaYAML := `#@data/values-schema
+---
+#@schema/examples key=True
+key: val
+`
+			expectedErr := `
+Invalid schema
+==============
+
+syntax error in @schema/examples annotation
+schema.yml:
+    |
+  3 | #@schema/examples key=True
+  4 | key: val
+    |
+
+    = found: keyword argument in @schema/examples (by schema.yml:3)
+    = expected: 2-tuple containing description (string) and example value (of expected type)
+`
+
+			filesToProcess := files.NewSortedFiles([]*files.File{
+				files.MustNewFileFromSource(files.NewBytesSource("schema.yml", []byte(schemaYAML))),
+			})
+
+			assertFails(t, filesToProcess, expectedErr, opts)
+		})
+		t.Run("does not match type of annotated node", func(t *testing.T) {
+			schemaYAML := `#@data/values-schema
+---
+#@schema/examples ("Zero value", 0)
+enabled: false
+`
+			expectedErr := `Invalid schema - @schema/examples has wrong type
+================================================
+
+schema.yml:
+    |
+  3 | #@schema/examples ("Zero value", 0)
+  4 | enabled: false
+    |
+
+    = found: integer (by schema.yml:3)
+    = expected: boolean (by schema.yml:4)
+    = hint: is the default value set using @schema/default?
```

Comment:
Just note: the hint is misplaced and is fixed in #582. This is a new instance of the hint and will need to be removed after _that_ PR is merged.